streams-runtimes/streams-runtime-threaded/src/main/java/org/apache/streams/threaded/controller/ThreadingController.java

```diff
@@ -144,6 +167,33 @@ public Integer getWorkingNow() {
     }
 
     /**
+     * Lock the ThreadingController. If the threading controller has not
+     * yet been used the thread pool for this controller will be allocated
+     * at this time.
+     */
+    public void lock() {
+        synchronized (this) {
```

Comment:
As above with the static methods, you could just put "synchronized" in the method signature and achieve the same behavior.